### DISCUSSION TITLE
atsar trouble installing

### DISCUSSION THREAD

**Discussion by mcmudge** [atsa-es/fish550-2023#31] [General]
<img width="835" alt="image" src="https://user-images.githubusercontent.com/43593981/235745384-9be2e1e9-d538-4bf3-ab66-dc6d37375f92.png">

When I try to download the atsar package, it gives me this error that additional tools are needed. If I click no, it fails. If I click yes, it fails with and says to call package build - but that sends me back to the popup seen here.

**Comment by mdscheuerell** [comment]
@ericward-noaa: I wonder if this is a symptom of outdated XCode?

**Comment by mcmudge** [comment]
I've tried checking XCode in Terminal by running "xcode-select --install" which gives me the message"xcode-select: error: command line tools are already installed, use "Software Update" in System Settings to install updates" but there aren't any software updates in system settings. So I have CommandLineTools. 

Here's more of the error message in R. My mac version is Ventura 13.3.1
<img width="981" alt="image" src="https://user-images.githubusercontent.com/43593981/235793170-aaee961e-c438-4827-b1ca-b031e2cc752c.png">

**Comment by mdscheuerell** [comment]
I found this [possible solution on SO](https://stackoverflow.com/questions/37776377/error-when-installing-an-r-package-from-github-could-not-find-build-tools-neces):

Copy/paste the following into your R console and run it. After doing so, try to reinstall {atsar}.

`options(buildtools.check = function(action) TRUE )`

**Comment by mcmudge** [comment]
That did help with installing the needed packages that I didn't originally have, but I still have an error: 
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/43593981/235804460-4fcf5139-d0b2-4cb0-8b4c-c99e35436115.png">

I looked up the error "wchar.h file not found" and an SO thread suggested that it may mean R doesn't know where to find Xcode. I'm not sure how to do this in Rstudio though. The top comment does have the correct location for my Xcode, but the sudo command doesn't work in R. https://stackoverflow.com/questions/26185978/macos-wchar-h-file-not-found

**Comment by mdscheuerell** [comment]
You'll have to run the sudo command from a Terminal, not the R console (you can copy/paste). To do so, you have 2 options:

1) Use the terminal in Rstudio; the tab is right next to your standard console window (see image below)

<img width="990" alt="Screen Shot 2023-05-02 at 4 20 09 PM" src="https://user-images.githubusercontent.com/5048528/235805692-8b0fe8f3-e57d-452e-bb6e-fc34afb1b99f.png">

2) Use the MacOS Terminal (press command+space to bring up the spotlight search and enter `terminal.app` if you can't find it)

<img width="678" alt="Screen Shot 2023-05-02 at 4 23 59 PM" src="https://user-images.githubusercontent.com/5048528/235806017-1dd52c1c-abb6-4453-a1f3-fade2f2a2fc7.png">

<img width="977" alt="Screen Shot 2023-05-02 at 4 22 51 PM" src="https://user-images.githubusercontent.com/5048528/235805924-c4f28ecb-9ce6-40f3-b240-ad82271bcf32.png">

**Comment by eeholmes** [comment]
Hi,

If you are on a Windows or MacOS (not M1), then you can install atsar from our ATSA-ES repository that has binaries. You shouldn't have to build the package unless you are on a M1/2 mac:

Here: https://atsa-es.r-universe.dev/atsar

To install
```
install.packages('atsar', repos = c('https://atsa-es.r-universe.dev', 'https://cloud.r-project.org'))
```

**Comment by mcmudge** [comment]
I jumped straight to Eli's comment which worked. Thanks Mark and Eli!
